circleguard/settings.py

```diff
@@ -140,38 +157,78 @@ def on_setting_changed_from_gui(self, value):
+        "template_steal":      ("[osu!std] {r.later_replay.username} | Replay Stealing"
+                                "\n\n"
+                                "Profile: https://osu.ppy.sh/users/{r.later_replay.user_id}"
+                                "\n\n"
+                                "Map: https://osu.ppy.sh/b/{r.later_replay.map_id}"
+                                "\n\n"
+                                "{r.later_replay.username}'s replay (cheated): https://osu.ppy.sh/scores/osu/{r.later_replay.replay_id}/download"
+                                "\n\n"
+                                "{r.earlier_replay.username}'s replay (original): https://osu.ppy.sh/scores/osu/{r.earlier_replay.replay_id}/download"
+                                "\n\n"
+                                "{r.similarity:.2f} similarity according to [circleguard](https://github.com/circleguard/circleguard) (higher is less similar)"),
+        "template_relax":      ("[osu!std] {r.replay.username} | Relax"
+                                "\n\n"
+                                "Profile: https://osu.ppy.sh/users/{r.replay.user_id}"
+                                "\n\n"
+                                "Map: https://osu.ppy.sh/b/{r.replay.map_id}"
+                                "\n\n"
+                                "replay download: https://osu.ppy.sh/scores/osu/{r.replay.replay_id}/download"
+                                "\n\n"
+                                "ur (ucv): {r.ur:.2f} according to [circleguard](https://github.com/circleguard/circleguard)"),
+        "template_correction": ("[osu!std] {r.replay.username} | Aim Correction"
+                                "\n\n"
+                                "Profile: https://osu.ppy.sh/users/{r.replay.user_id}"
+                                "\n\n"
+                                "Map: https://osu.ppy.sh/b/{r.replay.map_id}"
+                                "\n\n"
+                                "replay download: https://osu.ppy.sh/scores/osu/{r.replay.replay_id}/download"
+                                "\n\n"
+                                "Snaps according to [circleguard](https://github.com/circleguard/circleguard):"
+                                "\n\n"
+                                "{snap_table}")
     },
     "Thresholds": {
-        "threshold_cheat": 18,
-        "threshold_display": 25
+        "steal_max_sim": 18,
+        "steal_max_sim_display": 25,
+        "relax_max_ur": 50,
+        "relax_max_ur_display": 90,
+        "correction_max_angle": 10,
+        "correction_max_angle_display": 10,
+        "correction_min_distance": 8,
+        "correction_min_distance_display": 8
     },
     "Appearance": {
         "dark_theme": False,
         "visualizer_info": True,
-        "visualizer_bg": False
+        "visualizer_black_bg": False,
+        "required_style": "QLineEdit { border: 1px solid red }\n"
+                          "WidgetCombiner { border: 1px solid red }"
```

Comment:
This isn't a setting because I think people will want to change it, but more so because I need to access it at various points in the code (not just a single function) and making it a global is pretty nasty, so I made it a setting

Reply:
we really need a better solution for this but I guess it's okay for now.